bazel/crates.bzl

```diff
@@ -0,0 +1,113 @@
+"""
+cargo-raze crate workspace functions
+
+DO NOT EDIT! Replaced on runs of cargo-raze
+"""
+
+load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")
+load("@bazel_tools//tools/build_defs/repo:git.bzl", "new_git_repository")
+
+def _new_http_archive(name, **kwargs):
+    if not native.existing_rule(name):
+        http_archive(name = name, **kwargs)
+
+def _new_git_repository(name, **kwargs):
+    if not native.existing_rule(name):
+        new_git_repository(name = name, **kwargs)
+
+def raze_fetch_remote_crates():
+    _new_http_archive(
+        name = "raze__ahash__0_3_8",
+        url = "https://crates-io.s3-us-west-1.amazonaws.com/crates/ahash/ahash-0.3.8.crate",
+        type = "tar.gz",
+        strip_prefix = "ahash-0.3.8",
+        build_file = Label("//bazel/external/cargo/remote:ahash-0.3.8.BUILD"),
+    )
+
+    _new_http_archive(
+        name = "raze__autocfg__1_0_0",
+        url = "https://crates-io.s3-us-west-1.amazonaws.com/crates/autocfg/autocfg-1.0.0.crate",
+        type = "tar.gz",
+        strip_prefix = "autocfg-1.0.0",
+        build_file = Label("//bazel/external/cargo/remote:autocfg-1.0.0.BUILD"),
+    )
+
+    _new_http_archive(
+        name = "raze__cfg_if__0_1_10",
+        url = "https://crates-io.s3-us-west-1.amazonaws.com/crates/cfg-if/cfg-if-0.1.10.crate",
+        type = "tar.gz",
+        strip_prefix = "cfg-if-0.1.10",
+        build_file = Label("//bazel/external/cargo/remote:cfg-if-0.1.10.BUILD"),
+    )
+
+    _new_http_archive(
+        name = "raze__hashbrown__0_7_2",
+        url = "https://crates-io.s3-us-west-1.amazonaws.com/crates/hashbrown/hashbrown-0.7.2.crate",
+        type = "tar.gz",
```

Comment:
These are all very similar, can you create a `_rust_crate(..)` macro that factors out all the boilerplate? it should be possible to have just `_rust_crate(name="hashbrown", version="0.7.2")` and it can fill the rest I think.

Reply:
Those are all automatically generated by `cargo raze` tool, using macros here would turn this into manual work.